modules/core/src/main/java/org/apache/ignite/internal/causality/BaseVersionedValue.java

```diff
@@ -341,4 +369,37 @@ private void notifyCompletionListeners(long causalityToken, CompletableFuture<T>
             }
         });
     }
+
+    /**
+     * Deletes all versions from the Version Value up to and including the {@code causalityToken} and notify all
+     * {@link #whenDelete(DeletionListener) registered deletion listeners}.
+     *
+     * <p>{@code causalityToken} is expected to be less than the last {@link #complete completed} and greater than the last deleted.</p>
+     *
+     * @param causalityToken Causality token.
+     */
+    void delete(long causalityToken) {
```

Comment:
I think `removeUpTo` is a more consistent name for this method

Reply:
I think delete is more correct because it means to throw away forever, when remove means to throw away forever not immediately, maybe later.

Reply:
It should be called `deleteUpTo` then